internal/integrationtest/compile/compile_part_1_test.go

```diff
@@ -0,0 +1,507 @@
+
+	// Init the environment explicitly
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+
+	// Install Arduino AVR Boards
+	_, _, err = cli.Run("core", "install", "arduino:avr@1.8.3")
+	require.NoError(t, err)
+
+	sketchName := "RCS"
+	sketchPath := cli.SketchbookDir().Join(sketchName)
+	fqbn := "arduino:avr:uno"
+
+	// Create a test sketch
+	stdout, _, err := cli.Run("sketch", "new", sketchPath.String())
+	require.NoError(t, err)
+	require.Contains(t, string(stdout), "Sketch created in: "+sketchPath.String())
+
+	// Build sketch for arduino:avr:uno
+	_, _, err = cli.Run("compile", "-b", fqbn, sketchPath.String())
+	require.NoError(t, err)
+}
+
+func TestCompileWithoutPrecompiledLibraries(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	// Init the environment explicitly
+	url := "https://adafruit.github.io/arduino-board-index/package_adafruit_index.json"
+	_, _, err := cli.Run("core", "update-index", "--additional-urls="+url)
+	require.NoError(t, err)
+	_, _, err = cli.Run("core", "install", "arduino:mbed@1.3.1", "--additional-urls="+url)
+	require.NoError(t, err)
+
+	// // Precompiled version of Arduino_TensorflowLite
+	//	_, _, err = cli.Run("lib", "install", "Arduino_LSM9DS1")
+	//	require.NoError(t, err)
+	//	_, _, err = cli.Run("lib", "install", "Arduino_TensorflowLite@2.1.1-ALPHA-precompiled")
+	//	require.NoError(t, err)
+
+	//	sketchPath := cli.SketchbookDir().Join("libraries", "Arduino_TensorFlowLite", "examples", "hello_world")
+	//	_, _, err = cli.Run("compile", "-b", "arduino:mbed:nano33ble", sketchPath.String())
+	//	require.NoError(t, err)
+
+	_, _, err = cli.Run("core", "install", "arduino:samd@1.8.7", "--additional-urls="+url)
+	require.NoError(t, err)
+	//	_, _, err = cli.Run("core", "install", "adafruit:samd@1.6.4", "--additional-urls="+url)
+	//	require.NoError(t, err)
+	//	// should work on adafruit too after https://github.com/arduino/arduino-cli/pull/1134
+	//	_, _, err = cli.Run("compile", "-b", "adafruit:samd:adafruit_feather_m4", sketchPath.String())
+	//	require.NoError(t, err)
+
+	//	// Non-precompiled version of Arduino_TensorflowLite
+	//	_, _, err = cli.Run("lib", "install", "Arduino_TensorflowLite@2.1.0-ALPHA")
+	//	require.NoError(t, err)
+	//	_, _, err = cli.Run("compile", "-b", "arduino:mbed:nano33ble", sketchPath.String())
+	//	require.NoError(t, err)
+	//	_, _, err = cli.Run("compile", "-b", "adafruit:samd:adafruit_feather_m4", sketchPath.String())
+	//	require.NoError(t, err)
+
+	// Bosch sensor library
+	_, _, err = cli.Run("lib", "install", "BSEC Software Library@1.5.1474")
+	require.NoError(t, err)
+	sketchPath := cli.SketchbookDir().Join("libraries", "BSEC_Software_Library", "examples", "basic")
+	_, _, err = cli.Run("compile", "-b", "arduino:samd:mkr1000", sketchPath.String())
+	require.NoError(t, err)
+	_, _, err = cli.Run("compile", "-b", "arduino:mbed:nano33ble", sketchPath.String())
```

Comment:
This command is returning `exit status 1` when running on Windows and the same is happening on VSCode. You can check the failing workflow run here: https://github.com/arduino/arduino-cli/actions/runs/3066849086/jobs/4953246005#step:6:2274
On the other hand, when I try to reproduce it locally using the command prompt, it runs smoothly with no errors:
![image](https://user-images.githubusercontent.com/109663225/190640427-24a3e55b-3196-4412-9e03-8ce15592e2ab.png)
@cmaglie @umbynos do you have any suggestions?

Reply:
In your local machine you have `arduino:mbed` version `3.2.0`, instead in the CI environment is `1.3.1`. Also in the CI env it seems that the compiler is unable to find the `SPI` library... I don't know if this is correlated.

Reply:
It is related to the path length. The reason the test fails and the normal command is successful is that the temporary path set for `directories.data` by `github.com/arduino/arduino-cli/internal/integrationtest.CreateArduinoCLIWithEnvironment` is longer than the default `directories.data` path used when you ran that command. 

The normal `directories.data` path is like this:

```text
C:\Users\matteopologruto\Appdata\Local\Arduino15
```

The one used by the integration test is like [this](https://github.com/arduino/arduino-cli/actions/runs/3066849086/jobs/4953246005#step:6:2294):

```text
C:\Users\runneradmin\AppData\Local\Temp\go-testsuite625510244\Arduino15
```

Here is a demo:

Run the command with a relatively short `directories.data` path length and it is successful:

```text
$ export ARDUINO_DIRECTORIES_USER="/tmp/cli-dir/user"

$ export ARDUINO_DIRECTORIES_DATA="/tmp/cli-dir/data"

$ arduino-cli core install arduino:mbed@1.3.1

[...]

$ arduino-cli lib install "BSEC Software Library"@1.5.1474

[...]

$ arduino-cli compile -b arduino:mbed:nano33ble "${ARDUINO_DIRECTORIES_USER}/libraries/BSEC_Software_Library/examples/basic"

WARNING: library BSEC_Software_Library claims to run on samd, sam, esp8266, nrf52, esp32, avr architecture(s) and may be incompatible with your current board which runs on mbed architecture(s).
Library BSEC_Software_Library has been declared precompiled:
Precompiled library in "C:\Users\per\AppData\Local\Temp\cli-dir\user\libraries\BSEC_Software_Library\src\cortex-m4\fpv4-sp-d16-softfp" not found
Using precompiled library in C:\Users\per\AppData\Local\Temp\cli-dir\user\libraries\BSEC_Software_Library\src\cortex-m4
Sketch uses 139936 bytes (14%) of program storage space. Maximum is 983040 bytes.
Global variables use 48456 bytes (18%) of dynamic memory, leaving 213688 bytes for local variables. Maximum is 262144 bytes.

Used library          Version  Path
BSEC_Software_Library 1.5.1474 C:\Users\per\AppData\Local\Temp\cli-dir\user\libraries\BSEC_Software_Library
Wire                           C:\Users\per\AppData\Local\Temp\cli-dir\data\packages\arduino\hardware\mbed\1.3.1\libraries\Wire
SPI                            C:\Users\per\AppData\Local\Temp\cli-dir\data\packages\arduino\hardware\mbed\1.3.1\libraries\SPI

Used platform Version Path
arduino:mbed  1.3.1   C:\Users\per\AppData\Local\Temp\cli-dir\data\packages\arduino\hardware\mbed\1.3.1
```

Now change to a longer `directories.data` and it fails just like in the test:

```text
$ OLD_ARDUINO_DIRECTORIES_DATA="$ARDUINO_DIRECTORIES_DATA"

$ export ARDUINO_DIRECTORIES_DATA="/tmp/cli-dir/data-12345678901234567890123456789012345678901234567890"

$ mv "$OLD_ARDUINO_DIRECTORIES_DATA" "$ARDUINO_DIRECTORIES_DATA"

$ arduino-cli compile -b arduino:mbed:nano33ble "${ARDUINO_DIRECTORIES_USER}/libraries/BSEC_Software_Library/examples/basic"

In file included from c:\users\per\appdata\local\temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\tools\arm-none-eabi-gcc\7-2017q4\arm-none-eabi\include\c++\7.2.1\stdlib.h:36:0,
                 from C:\Users\per\AppData\Local\Temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\hardware\mbed\1.3.1\cores\arduino/api/String.h:27,
                 from C:\Users\per\AppData\Local\Temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\hardware\mbed\1.3.1\cores\arduino/api/Print.h:24,
                 from C:\Users\per\AppData\Local\Temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\hardware\mbed\1.3.1\cores\arduino/api/Stream.h:25,
                 from C:\Users\per\AppData\Local\Temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\hardware\mbed\1.3.1\cores\arduino/api/HardwareI2C.h:22,
                 from C:\Users\per\AppData\Local\Temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\hardware\mbed\1.3.1\libraries\Wire/Wire.h:22,
                 from C:\Users\per\AppData\Local\Temp\cli-dir\user\libraries\BSEC_Software_Library\src/bsec.h:44,
                 from C:\Users\per\AppData\Local\Temp\cli-dir\user\libraries\BSEC_Software_Library\examples\basic\basic.ino:1:
c:\users\per\appdata\local\temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\tools\arm-none-eabi-gcc\7-2017q4\arm-none-eabi\include\c++\7.2.1\cstdlib:41:10: fatal error: bits/c++config.h: No such file or directory
 #include <bits/c++config.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.


Used library          Version  Path
BSEC_Software_Library 1.5.1474 C:\Users\per\AppData\Local\Temp\cli-dir\user\libraries\BSEC_Software_Library
Wire                           C:\Users\per\AppData\Local\Temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\hardware\mbed\1.3.1\libraries\Wire

Used platform Version Path
arduino:mbed  1.3.1   C:\Users\per\AppData\Local\Temp\cli-dir\data-12345678901234567890123456789012345678901234567890\packages\arduino\hardware\mbed\1.3.1

Error during build: exit status 1
```

Reply:
>In your local machine you have arduino:mbed version 3.2.0, instead in the CI environment is 1.3.1. Also in the CI env it seems that the compiler is unable to find the SPI library... I don't know if this is correlated.

I also went down those rabbit holes @cmaglie, but it is not related. The results are the same even when using `arduino:mbed@3.2.0`. It is only about the path length.

The question is: what is the source of this inability to handle reasonable path lengths? Is it in the Arduino CLI codebase, the `arduino:mbed` platform, just Windows being trash as usual, or...?

Reply:
I have seen real world user report of this error and could never figure out what caused it.

Reply:
Sounds great, thanks for helping @cmaglie @per1234 ! I'll let you know if this issue is fixed once the patch is out 🤞

Reply:
Ok, actually I've already did it in perspective, just forgot it :-).
You can actually add the following lines in `integrationtest/arduino-cli.go`:

```go
func init() {
	testsuite.ProjectName = "cli"
}
```

This will reduce the name of the temp folder from `go-testsuite12345678` to `cli12345678`, actually shaving 9 characters from the path... hopefully this will be sufficient to let the test pass 🤞🏼

Reply:
Reducing path length worked, the tests are now successful on Windows. Changing the temp folder's name from `go-testsuite` to `cli` wasn't enough, so we decided to change `DataDir`'s name to `A`.